dockerworker/artifact_test.go

```diff
@@ -0,0 +1,36 @@
+// +build docker
+
+package dockerworker
+
+import (
+	"io"
+	"testing"
+
+	"github.com/mattetti/filebuffer"
+	"github.com/stretchr/testify/require"
+)
+
+func TestArtifact(t *testing.T) {
+	d := NewTestDockerWorker(t)
+
+	content := "This is a dummy artifact"
+
+	taskID, err := createDummyArtifact(d, "public/test", content)
+	require.NoError(t, err)
+
+	// Test Download Latest
+	buff := filebuffer.New([]byte{})
+	require.NoError(t, d.DownloadArtifact(taskID, "", "public/test", buff))
+
+	_, err = buff.Seek(0, io.SeekStart)
+	require.NoError(t, err)
+	require.Equal(t, content, buff.String())
+
+	// Test Download specific runID
+	buff = filebuffer.New([]byte{})
+	require.NoError(t, d.DownloadArtifact(taskID, "0", "public/test", buff))
+
+	_, err = buff.Seek(0, io.SeekStart)
+	require.NoError(t, err)
+	require.Equal(t, content, buff.String())
+}
```

Comment:
As far as I can see, the docker engine doesn't support artifacts yet. There is no way to mount artifacts, or extract them. Yet this test tests uploading and downloading artifacts, using the methods you wrote, which we established weren't needed, since generic-worker already provides methods to upload/download artifacts.